frmts/dimap/dimapdataset.cpp

```diff
@@ -1550,6 +1551,33 @@ int DIMAPDataset::ReadImageInformation2()
         CSLDestroy(papszRPC);
     }
 
+    CPLXMLNode *psLocatedUseAreaNode =
+        CPLGetXMLNode(
+            psDoc,
+            "Geometric_Data.Use_Area");
+    if(psLocatedUseAreaNode!=nullptr) {
+        CPLXMLNode *psLocatedGeometricValuesNode = psLocatedUseAreaNode->psChild;
+        while (psLocatedGeometricValuesNode != nullptr)
+        {
+            CPLXMLNode *psLocationType = CPLGetXMLNode(psLocatedGeometricValuesNode, "LOCATION_TYPE");
+            if (psLocationType==nullptr || !EQUAL(psLocationType->psChild->pszValue, "center"))
```

Comment:
for robustness against hostile datasets:
```suggestion
            if (psLocationType==nullptr || psLocationType->psChild == nullptr || !EQUAL(psLocationType->psChild->pszValue, "center"))
```